types/src/v0/impls/state.rs

```diff
@@ -663,6 +695,41 @@ impl HotShotState<SeqTypes> for ValidatedState {
             .resolve()
             .expect("Chain Config not found in validated state");
 
+        // Validate l1_finalized.
+        let Some(proposed_finalized) = proposed_header.l1_finalized() else {
+            tracing::error!("L1 finalized has None value.");
+            return Err(BlockError::InvalidBlockHeader);
+        };
+
+        let parent_finalized = parent_leaf.block_header().l1_finalized().unwrap().number();
```

Comment:
Since we're not adding tests now please open an issue for that (if we don't have it already) and link it in the code.

Reply:
0becffbc6e88029fa5afec89a8bd0f94ae05390e